frontend/src/pages/overview/overview/ChainDetails.js

```diff
@@ -115,121 +143,228 @@ export const ChainDetails = ({ chainID, chainName }) => {
+                  {originDenom}
+                </StyledTableCell>
+                <StyledTableCell>
+                  {(+staked / 10 ** decimals).toLocaleString()}&nbsp;
+                  {originDenom}
+                </StyledTableCell>
+                <StyledTableCell>
+                  {(+totalRewards / 10 ** decimals).toLocaleString()}&nbsp;
+                  {originDenom}
+                </StyledTableCell>
+                <StyledTableCell>
+                  {tokensPriceInfo[originMinimalDenom]
+                    ? `$${parseFloat(
+                        tokensPriceInfo[originMinimalDenom]?.info?.["usd"]
+                      ).toFixed(2)}`
+                    : "N/A"}
+                </StyledTableCell>
+                <StyledTableCell>
+                  <Button
+                    color="primary"
+                    disableElevation
+                    variant="contained"
+                    size="small"
+                    sx={{
+                      textTransform: "none",
+                    }}
+                    disabled={
+                      totalRewards <= 0 ||
+                      txRestakeStatus === "pending" ||
+                      distTxStatus?.status === "pending"
+                    }
+                    onClick={actionClaimAndStake}
+                  >
+                    {txRestakeStatus?.status === "pending" ? (
+                      <>
+                        <CircularProgress size={18} />
+                        &nbsp;Claim&nbsp;&&nbsp;Stake
+                      </>
+                    ) : (
+                      <>Claim&nbsp;&&nbsp;Stake</>
+                    )}
+                  </Button>
+                  <Button
+                    color="primary"
+                    disableElevation
+                    variant="contained"
+                    size="small"
+                    sx={{
+                      textTransform: "none",
+                      ml: 1,
+                    }}
+                    disabled={
+                      totalRewards <= 0 ||
+                      txRestakeStatus === "pending" ||
+                      distTxStatus?.status === "pending"
+                    }
+                    onClick={claimRewards}
+                  >
+                    {distTxStatus?.status === "pending" ? (
+                      <>
+                        <CircularProgress size={18} />
+                        &nbsp;Claim
+                      </>
+                    ) : (
+                      <>Claim</>
+                    )}
+                  </Button>
+                </StyledTableCell>
+              </StyledTableRow>
+            ) : null}
+          </>
+        </>
+      ) : (
+        <>
+          <>
+            {balance?.map((item, index) => {
+              const denomInfo = chainDenoms[chainName].filter((x) => {
+                return x.denom === item.denom;
+              });
+              dispatch(getTokenPrice(denomInfo.origin_denom));
```

Comment:
```suggestion
```
This is not required. In the main component we are fetching all the available tokens price information.